polkadot/roadmap/implementers-guide/src/runtime/parainherent.md

```diff
@@ -60,3 +60,32 @@ processing it, so the processed inherent data is simply dropped.
 This also means that the `enter` function keeps data around for no good reason. This seems acceptable though as the size
 of a block is rather limited. Nevertheless if we ever wanted to optimize this we can easily implement an inherent
 collector that has two implementations, where one clones and stores the data and the other just passes it on.
+
+## Data sanitization
+`ParasInherent` performs sanitization on the provided input data. When the module is invoked via `create_inherent` the
+input data is sanitized. `enter` entry point requires sanitized input. If unsanitized data is provided the module
+generates an error.
+
+Disputes are included in the block with a priority for a security reasons. It's important to pump as many dispute votes
+onchain as possible so that disputes conclude faster and the offenders are punished. However if there are too many
+disputes to include in a block the dispute set is trimmed so that it respects max block weight.
+
+Dispute data is first deduplicated and sorted by block number (older first) and dispute location (local then remote).
+Concluded and ancient (disputes initiated before the post conclusion acceptance period ) disputes are filtered out.
+Votes with invalid signatures and from unknown validators (not found in the active set for the current session) are also
+filtered out.
+
+All dispute statements are included included in the order described in the previous paragraph until the available block
```

Comment:
```suggestion
All dispute statements are included in the order described in the previous paragraph until the available block
```

Reply:
Actually this whole sentence seems redundant as you already state the same fact 2 paragraphs before:

> However if there are too many disputes to include in a block the dispute set is trimmed so that it respects max block weight.